mala/interfaces/ase_calculator_uncertainty.py

```diff
@@ -0,0 +1,157 @@
+"""ASE calculator for MALA predictions."""
+
+from ase.calculators.calculator import Calculator, all_changes
+import numpy as np
+
+from mala import Parameters, Network, DataHandler, LDOS, MultiPredictor
+from mala.common.parallelizer import get_rank, get_comm, barrier
+from mala.interfaces.ase_calculator import MALA
+
+
+class MALAUncertainty(MALA):
+    """
+    Implements an ASE calculator based on MALA predictions.
+
+    With this, MD simulations can be performed.
+
+    Parameters
+    ----------
+    params : mala.common.parametes.Parameters
+        Parameters used to create this Interface.
+
+    networks : list
+        List of networks to be used for the predictions.
+
+    data : mala.datahandling.data_handler.DataHandler
+        DataHandler; not really used in the classical sense, more as a
+        collector of properties and objects (such as a target calculator).
+
+    reference_data : list
+        A list containing
+
+        - [0]: A type of additional calculation data
+        - [1]: A path to it.
+
+        With this additonal calculation data (preferably from the training of
+        the neural network), calculator can access all important data such as
+        temperature, number of electrons, etc. that might not be known simply
+        from the atomic positions.
+    """
+
+    implemented_properties = ['energy', 'forces']
+
+    def __init__(self, params: Parameters, networks: list,
+                 data: DataHandler, reference_data=None,
+                 predictor=None):
+        self.mala_parameters: Parameters = params
+        if self.mala_parameters.targets.target_type != "LDOS":
+            raise Exception("The MALA calculator currently only works with the"
+                            "LDOS.")
+
+        if predictor is None:
+            predictor = MultiPredictor(params, networks, data)
+        else:
+            predictor = predictor
+
+        super(MALAUncertainty, self).__init__(params, networks[0],
```

Comment:
`super().__init__(...)` does it.

Reply:
This is what my IDE gives by default - I was under the impression that doing so is preferred because it is safe even for instances where there are multiple parent classes?

Reply:
It is valid code. However, in the vast majority of cases, especially in simple single inheritance, `super()` is usually preferred as it will produce the same method resolution order (MRO) as the more verbose `super(DerivedClass, self)` does, but it makes code more readable and maintainable, e.g. should `DerivedClass` be renamed. But it is a matter of style.

In complicated multiple inheritance cases, one may pass arguments to `super()` in order to influence the MRO. See also https://docs.python.org/3/library/functions.html#super .